plotbitrate.py

```diff
@@ -28,283 +29,619 @@
+              "--downscale has no effect" + ConsoleColors.END_COLOR)
+    
+    # check if downscale and show-frame-types are both given
+    if arguments.downscale and arguments.show_frame_types:
+        sys.exit(ConsoleColors.ERROR +
+                 "Error: Options --downscale and --show-frame-types cannot "
+                 "be given both")
+
+    arguments_dict = vars(arguments)
+
+    # set ffprobe stream specifier
+    if arguments.stream == "audio":
+        arguments_dict["stream_spec"] = "a"
+    elif arguments.stream == "video":
+        arguments_dict["stream_spec"] = "V"
+    else:
+        raise RuntimeError("Invalid stream type")
+
+    return arguments
+
+
+def open_ffprobe_get_format(file_path: str) -> subprocess.Popen:
+    """
+    Opens an ffprobe process that reads the format data
+    for file_path and returns the process.
+    """
+    return subprocess.Popen(
         ["ffprobe",
-            "-show_entries", "format",
-            "-print_format", "xml",
-            args.input
-        ],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.DEVNULL) as proc_format:
-
-        # parse format header xml
-        format_data = etree.parse(proc_format.stdout)
-        format_elem = format_data.find('.//format')
-
-        # save the total time for later
-        try:
-            total_time = float(format_elem.get('duration'))
-        except:
-            sys.stderr.write("Error: Failed to determine stream duration\n")
-            sys.exit(1)
-
-# set ffprobe stream specifier
-if args.stream == 'audio':
-    stream_spec = 'a'
-elif args.stream == 'video':
-    stream_spec = 'V'
-else:
-    raise RuntimeError("Invalid stream type")
-
-# get frame data for the selected stream
-with subprocess.Popen(
-    ["ffprobe",
-        "-show_entries", "frame",
-        "-select_streams", stream_spec,
-        "-print_format", "xml",
-        args.input
-    ],
-    stdout=subprocess.PIPE,
-    stderr=subprocess.DEVNULL) as proc_frame:
-
-    # process xml elements as they close
-    for event in etree.iterparse(proc_frame.stdout):
-
-        # skip non-frame elements
-        node = event[1]
-        if node.tag != 'frame':
+         "-hide_banner",
+         "-loglevel", "error",
+         "-show_entries", "format",
+         "-print_format", "xml",
+         file_path
+         ],
+        stdout=subprocess.PIPE)
+
+
+def open_ffprobe_get_frames(
+        file_path: str,
+        stream_selector: str
+) -> subprocess.Popen:
+    """
+    Opens an ffprobe process that reads all frame data for
+    file_path and returns the process.
+    """
+    return subprocess.Popen(
+        ["ffprobe",
+         "-hide_banner",
+         "-loglevel", "error",
+         "-select_streams", stream_selector,
+         "-threads", str(multiprocessing.cpu_count()),
+         "-print_format", "xml",
+         "-show_entries",
+         "frame=pict_type,pkt_duration_time,pkt_pts_time," +
```

Comment:
We don't use the `pkt_duration_time` field anymore so we can remove it from here and the subprocess instance below. That should save some disk space and speed up processing ever so slightly.